packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -77,20 +74,24 @@ export class BlobManager {
             `${BlobManager.basePath}/${response.id}`,
             this.routeContext,
             async () => this.getStorage().readBlob(response.id),
-            () => this.attachBlobCallback(response.id),
         );
 
         // Note - server will de-dup blobs, so we might get existing blobId!
         if (!this.blobIds.has(response.id)) {
-            this.pendingBlobIds.add(response.id);
+            this.pendingBlobIds.set(response.id, new Deferred<void>());
+
+            // send blob attach op and wait until we see it to return the handle
+            this.attachBlobCallback(response.id);
+            await this.pendingBlobIds.get(response.id)?.promise;
```

Comment:
Not sure if we want to do it in this PR, but we need to check the policy (returned as payload of upload API in ODSP) if blob op made it in time. If not, we would need to restart the uploads process. If not in this PR, can you please check we have an issue tracking it (and if not - open one)?
Also we should likely reject process (overall end-to-end) if container runtime was disposed.

Reply:
#6469

Reply:
What about rejection on container being disposed? That's missing and not covered in this issue.

Reply:
submit() will throw if the runtime is disposed and I added a check when uploadBlob() is called. Do we need more?

Reply:
I think you refer to attachBlobCallback() callback calling submit and failing if container runtime is disposed at that moment.
This is not sufficient. The biggest gap - something somewhere needs to do this.pendingBlobIds.get(id).reject() when container runtime is disposed or we should have Promise.race() waiting for promise that gets rejected on disposition of container runtime.
The other part - failing before making a call to createBlob(). I've looked at ODSP implementation and we do not track dispose there, so it will be happy to create a blob in disposed state, which is not great (but less important than above)